flexget/plugins/input/telegram_input.py

```diff
@@ -0,0 +1,228 @@
+
+        #We have a error
+        if not response['ok']:
+            logger.error("Telegram updater returned error {}: {}", response['error_code'], response['description'] )
+            return
+
+        #Get All New Messages
+        messages = response['result']
+
+        entries = []
+        for message in messages:
+            #This is the ID
+            update_id = message['update_id']
+
+            #If desired update the last ID for the Bot
+            if clean_update:
+                logger.debug("Last Update set to {}",update_id)
+                task.simple_persistence[token+'_update_id'] = update_id
+            
+            #We Don't care if it's not a message or no text
+            if 'message' not in message:
+                continue
+
+            if 'text' not in message['message']:
+                continue
+
+            if 'chat' not in message['message']:
+                continue
+
+            if 'type' not in message['message']['chat']:
+                continue
+
+            logger.debug("Income message: {}", message)
+
+            #Create Base Entry
+            text = message['message']['text']
+            entry = Entry()
+            entry['title'] =  text
+
+            #We need a url, so we add a dummy
+            entry['url'] = "http://localhost?update_id="+str(update_id)
+
+            #Store the message if we need to use it in other plugins
+            entry['telegram_message'] =  message['message']
+            
+            #Check Types
+            if types:
+                if message['message']['chat']['type'] not in types:
+                    logger.debug("Ignoring message because of invalid type {}",message)
+                    continue
+
+            #Check From
+            append = True
+            if whitelist:
+                append = False
+                for i in whitelist:
+                    if 'username' in i and i['username'] == message['message']['from']['username']:
+                        logger.debug("WhiteListing: Username {}",message['message']['from']['username'])
+                        append = True
+                        break
+                    elif ('fullname' in i) and (i ['fullname']['first'] == message['message']['from']['first_name'] and i['fullname']['sur'] == message['message']['from']['last_name']):
+                        logger.debug("WhiteListing: Full Name {} {}",message['message']['from']['first_name'],message['message']['from']['last_name'])
+                        append = True
+                        break
+                    elif 'group' in i:
+                        if message['message']['chat']['type'] == 'group':
+                            if message['message']['chat']['title'] == i['group']:
+                                logger.debug("WhiteListing: Group {}",message['message']['chat']['title'])
+                                append = True
+                                break
+
+                if not append:
```

Comment:
instead of using a control variable, you can use the `else` clause for the `for` loop that will be called if it reached its end (without breaking or returning earlier)